dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/SignIn/Register/CheckAnswers.cshtml.cs

```diff
@@ -0,0 +1,88 @@
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using TeacherIdentity.AuthServer.Models;
+using TeacherIdentity.AuthServer.Pages.SignIn.Trn;
+
+namespace TeacherIdentity.AuthServer.Pages.SignIn.Register;
+
+public class CheckAnswers : PageModel
+{
+    private readonly IClock _clock;
+    private readonly TrnLookupHelper _trnLookupHelper;
+    private readonly TeacherIdentityServerDbContext _dbContext;
+    private readonly IdentityLinkGenerator _linkGenerator;
+
+    public CheckAnswers(
+        IdentityLinkGenerator linkGenerator,
+        TeacherIdentityServerDbContext dbContext,
+        IClock clock,
+        TrnLookupHelper trnLookupHelper)
+    {
+        _linkGenerator = linkGenerator;
+        _dbContext = dbContext;
+        _clock = clock;
+        _trnLookupHelper = trnLookupHelper;
+    }
+
+    public string BackLink => HttpContext.GetAuthenticationState().OAuthState?.RequiresTrnLookup == true
+        ? _linkGenerator.RegisterIttProvider()
```

Comment:
This should work as each page acts as a filter and redirects to the previous if HasXSet is false/null

Reply:
Fine for now but we'll have to review when the TRN lookup bits are in place (if a TRN is found early then the journey is 'short-circuited' so not all questions may have been asked).